src/app/services/shared-activated-route/shared-activated-route.service.ts

```diff
@@ -134,12 +134,21 @@ export class SharedActivatedRouteService extends withUnsubscribe() {
   private observeRouterEvents(router: Router, route: ActivatedRoute): void {
     /** Recursively find all children, and return in order with observable */
     const expandAllChildren = (): Observable<ActivatedRoute> => {
+      // Set a maximum depth to check. The spectator ActivatedRouteStub has
+      // infinite children, which causes errors during tests
+      const maxDepth = 5;
+      let depth = 0;
```

Comment:
this should probably be a weakref list of instances visited to avoid the recursion... depth limits in recursion are always code smells